testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveViews.java

```diff
@@ -172,5 +178,62 @@ public void testLateralViewJsonTupleAs()
                 "SELECT * FROM test_json_tuple_view",
                 queryAssert -> queryAssert.containsOnly(row(3, "Mateusz", "Gajewski", "true", "1000", null, null)));
     }
+
+    @Test(groups = HIVE_VIEWS)
+    public void testUnionAllViews()
+    {
+        onHive().executeQuery("DROP VIEW IF EXISTS union_helper");
+        onHive().executeQuery("CREATE VIEW union_helper AS\n"
+                + "SELECT (n_regionkey % 3) AS n_regionkey FROM nation");
+
+        onHive().executeQuery("DROP VIEW IF EXISTS union_all_view");
+        onHive().executeQuery("CREATE VIEW union_all_view AS\n"
+                + "SELECT n_regionkey FROM nation\n"
+                + "UNION ALL\n"
+                + "SELECT n_regionkey FROM union_helper\n");
+
+        assertThat(query("SELECT n_regionkey FROM union_all_view"))
+                // Copy the keys 5 times because there are 5 nations per region
+                .containsOnly(flatten(nCopies(5, asRows(List.of(
```

Comment:
> `flatten(nCopies(5, asRows(List.of`

i can imagine simpler assertions that that.

Maybe 

- add a WHERE condition  in union_helper so it produces just one duplicate 
- add a WHERE condition to nation table (or switch to smaller region), so that it produces just few rows (with a duplicate)
- have `containsOnly` with explicit `row` values, without any functions:


```
containsOnly(
  // from union_helper
   row(1),
   row(2),
   row(0),
    row(1),
 // from nation with a filter
     row(1),
    row(2),
   row(3));

Reply:
I don't know why I didn't just use the region table in the first place. Switched.